utils/common/src/main/java/org/apache/brooklyn/util/javalang/coerce/TypeCoercerExtensible.java

```diff
@@ -209,12 +226,19 @@ public static TypeCoercerExtensible newEmpty() {
                     try {
                         return Maybe.of((T) m.invoke(null, value));
                     } catch (Exception e) {
-                        Maybe.absent(new ClassCoercionException("Cannot coerce type "+value.getClass()+" to "+targetType.getCanonicalName()+" ("+value+"): "+m.getName()+" adapting failed, "+e));
+                        exceptions.add(new ClassCoercionException("Cannot coerce type "+value.getClass()+" to "+targetType.getCanonicalName()+" ("+value+"): "+m.getName()+" adapting failed", e));
                     }
                 }
             }
         }
-        return null;
+        if (exceptions.isEmpty()) {
+            return null;
```

Comment:
Worth adding a comment somewhere on the notation (caller):
  * null - no errors, continue with fallbacks (i.e. not found)
  * absent - had some kind of exception, continue with fallbacks
  * present - coercion successful, return value